### PR TITLE
[Dep] Add PHPUnit\Framework\TestCase stub on target-repository bootstrap.php

### PR DESCRIPTION
**Pull request by samsonasik**
Alternative of https://github.com/rectorphp/rector-symfony/pull/125

Closes https://github.com/rectorphp/rector-symfony/pull/125

### CODE REVIEW

**Comment by samsonasik** [comment]
@bocharsky-bw I verified this alternative at https://github.com/bocharsky-bw/no-phpunit-rector-reproducer repository.

**Comment by bocharsky-bw** [comment]
I'm fine with either way of solving this. If it won't cause errors when the project does not have PHPUnit installed - great 👍 

Thank you for working on it!

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: build/target-repository/bootstrap.php [hunk 1]
```diff
@@ -2,69 +2,80 @@
...
+}
+
+namespace PHPUnit\Framework
+{
+    if (! class_exists('PHPUnit\Framework\TestCase')) {
+        abstract class TestCase
+        {
+        }
     }
 }
```
Comment (TomasVotruba) [inline build/target-repository/bootstrap.php:82]: Could you link the original issue, so we know why we use this hack?

Comment (samsonasik) [reply build/target-repository/bootstrap.php:82]: added https://github.com/rectorphp/rector-src/pull/1924/commits/553d54b6a5b4fb8416b5e06a048f3a9d2a1be06b for `@see` to original issue.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Comment by bocharsky-bw** [comment]
Confirm that `rector/rector:dev-main` works for me now, thank you for this fix!

**Pull request merged**